src/OrchardCore.Modules/OrchardCore.Media/GraphQL/MediaAssetQuery.cs

```diff
@@ -0,0 +1,62 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GraphQL.Resolvers;
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Primitives;
+using OrchardCore.Apis.GraphQL;
+using OrchardCore.FileStorage;
+
+namespace OrchardCore.Media.GraphQL
+{
+    public class MediaAssetQuery : ISchemaBuilder
+    {
+        public MediaAssetQuery(IStringLocalizer<MediaAssetQuery> localizer)
+        {
+            T = localizer;
+        }
+
+        public IStringLocalizer T { get; set; }
+
+        public Task<IChangeToken> BuildAsync(ISchema schema)
+        {
+            var field = new FieldType
+            {
+                Name = "MediaAssets",
+                Description = T["Media assets are items that are part of your media library."],
+                Type = typeof(ListGraphType<MediaAssetObjectType>),
+                Arguments = new QueryArguments(
+                    new QueryArgument<StringGraphType>
+                    {
+                        Name = "path",
+                        Description = T["Media asset path."]
+                    },
+                    new QueryArgument<BooleanGraphType>
+                    {
+                        Name = "includeSubDirectories",
+                        Description = T["Whether to get the assets from just the top directory or from all sub-directories as well."]
+                    }
+                ),
+                Resolver = new AsyncFieldResolver<IEnumerable<IFileStoreEntry>>(ResolveAsync)
+            };
+
+            schema.Query.AddField(field);
+
+            return Task.FromResult<IChangeToken>(null);
+        }
+
+        private async Task<IEnumerable<IFileStoreEntry>> ResolveAsync(ResolveFieldContext resolveContext)
+        {
+            var context = (GraphQLContext)resolveContext.UserContext;
```

Comment:
Do we need to check the permission here?
Also, there is a special folder with content-local assets, this folder should be black-listed explicitly.  /cc @matiasmolleja

Reply:
I considered query level permission checks but it's not something we've done before for GraphQL. There also wasn't a 'ReadMedia' permission and I figured if the user has 'ExecuteGraphQL' permissions that allows them to read all content then allowing them to enumerate the list of media is probably fine.
If we want more granular control then I'm happy to implement a QueryMedia permission but query level permissions probably needs more thought and a general design?

I'm not familiar with this special folder, do you know the name so I can have a look?